client/src/views/Mapping/components/FhirMappingPanel/components/ResourceSelector/index.tsx

```diff
@@ -1,163 +1,78 @@
-import {
-  FormGroup,
-  ControlGroup,
-  Button,
-  InputGroup,
-  Spinner,
-  Icon
-} from "@blueprintjs/core";
-import * as QueryString from "query-string";
+import { FormGroup, ControlGroup, Button } from "@blueprintjs/core";
 import * as React from "react";
-import { Mutation } from "react-apollo";
 import { useSelector, useDispatch } from "react-redux";
 import useReactRouter from "use-react-router";
 
-import {
-  deselectFhirResource,
-  updateFhirResource
-} from "../../../../../../services/selectedNode/actions";
+import { updateLocationParams } from "src/services/urlState";
+import { updateFhirResource } from "src/services/selectedNode/actions";
+import { IReduxStore } from "src/types";
 
-import ResourceSelect from "../../../../../../components/selects/resourceSelect";
-import { IReduxStore } from "../../../../../../types";
+import ResourceSelect from "src/components/selects/resourceSelect";
 
-const deleteResourceMutation = require("../../../../../../graphql/mutations/deleteResource.graphql");
-const updateResource = require("../../../../../../graphql/mutations/updateResource.graphql");
+import Drawer from "../Drawer";
```

Comment:
nit: use absolute imports

Reply:
I deliberately chose to leave this one relative, for the following reason : `ResourceSelector` and `Drawer` are part of the same component `FhirMappingPanel`. The latter might later be moved to a different location (imagine it is now used in a second view : I would probably move it to `./src/components`). In this component, relative paths are immune to relocating the main component `FhirMappingPanel`, whereas absolute paths are not.
I'd like your opinion on that - rather unimportant, I do admit it - matter ; what do you think? :thinking:

Reply:
"Forward" relative imports seem ok to me, " strictly backwards" should probably be forbidden, but I don't know how to deal with "juxtaposed" components.

Reply:
`Drawer` should probably go explicitly inside `ResourceSelector` actually...

Reply:
Ok I get the relative import (makes sense).
About your question: "juxtaposed" components should be used in case `Drawer` is shared by several other components, otherwise it should indeed go inside `ResourceSelector`.

Just a word on this (not for this PR, just a thought): I think you can drop the "components" folders in your repo structure. eg: `client/src/views/Mapping/FhirMappingPanel/ResourceSelector/index.tsx `

Reply:
You are right, I moved `Drawer` to the right place in 36b2345.

About this `components` folder thing, I guess you are right too. I was following this article's recommendations: https://medium.com/@alexmngn/how-to-better-organize-your-react-applications-2fd3ea1920f1
I guess it is only useful to have a `components` folder if you also have a `services` or `views` folder in the same place, otherwise it's just too verbose. I'll try and do this in the future :slightly_smiling_face: